app/controllers/nodes_controller.rb

```diff
@@ -18,6 +18,61 @@ def show
     @minion = Minion.find(params[:id])
   end
 
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/PerceivedComplexity
+  def assign_roles
+    minion = Minion.find(params[:master_id])
+    minion_errors = []
+
+    if minion.assign_role(:master)
```

Comment:
Now that we have other requirements, the assign_roles method in app/models/minion.rb is not that useful. I would try to refactor that method to accept a Hash as the `roles` argument so it can handle this case too.

E.g.

```
  Minion.assign_roles({master: ["master.example.com"], minion: ["must_be_minion.example.com"]}, default_role: :minion)
```

This way we keep the controller thin and that method becomes useful again.